### PR TITLE
rewrite of the docs

### PR DESCRIPTION
**Pull request by jgrewe**
this pr is quite extensive, it replaces the former tutorials and updates all used scripts to 1.5 library version.
this pr further fixes a few docstring glitches  and adds a few more.

### CODE REVIEW

File: docs/source/storing_data.rst [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+type and the dimensionality (i.e. the number of dimensions) are fixed
+once the *DataArray* has been created. The actual size of the
+*DataArray* can be changed during the life-time of the entity.
+
+In case you need more control, *DataArrays* can be created empty for
+later filling e.g. during data acquisition.
+
+.. code-block:: python
+
+    data_array = block.create_data_array("sinewave", "nix.sampled", dtype=nix.DataType.Double, shape=(100));
```
Comment (mpsonntag) [inline docs/source/storing_data.rst:82]:
```suggestion
    data_array = block.create_data_array("sinewave", "nix.sampled", dtype=nixio.DataType.Double, shape=(100));
```

File: docs/source/storing_data.rst [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+
+   # replacing data
+   array[:100] = np.zeros(100)
+
+*DataArrays* can also be extended by ``appending`` data.
+
+.. code-block:: python
+
+   data_array = block.create_data_array("test data", "test", data=np.random.randn(100))
+   print(array.shape)
```
Comment (mpsonntag) [inline docs/source/storing_data.rst:113]: Should `array` in this example be `data_array`?

File: docs/source/storing_data.rst [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+
+.. literalinclude:: examples/irregularlySampledData.py
+   :caption: Storing irregularly sampled data (:download:`example code <examples/irregularlySampledData.py>`).
+   :language: python
+   :lines: 45 - 58
+   :emphasize-lines: 14
+
+**Note:** The *ticks* of a *RangeDimension* must be numeric and ascending.
+
+The *RangeDimension* can do a more. Consider the case that the times of an event are stored:
```
Comment (mpsonntag) [inline docs/source/storing_data.rst:196]:
```suggestion
The *RangeDimension* can do more. Consider the case that the times of an event are stored:
```

File: docs/source/storing_data.rst [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+For example these might be the times of action potentials (aka spikes) recorded in a nerve. In such a case it is basically the x-values that are of interest. It would be inefficient to store them twice, first as values in the *DataArray* and then again as ticks in the dimension descriptor.
+In such cases the *RangeDimension* is set up to link to the *DataArray* itself.
+
+.. literalinclude:: examples/rangeDimensionLink.py
+   :lines: 12 - 17
+   :language: python
+   :caption: A *RangeDimension* can link to a *DataArray* as a source of the ticks (:download:`exmaple code <examples/rangeDimensionLink.py>`). 
+   :emphasize-lines: 6
+
+In the highlighted line we use a convenience function to establish the link between the dimension descriptor and the
```
Comment (mpsonntag) [inline docs/source/storing_data.rst:210]:
```suggestion
In the highlighted line we use a convenience function to establish the link between the dimension descriptor and the data stored via the dimensions of its own DataArray.
```
The sentence is missing its end, attempting to complete.

File: docs/source/storing_data.rst [hunk 5]
```diff
@@ -0,0 +1,382 @@
...
+creation. Available flags are:
+
+-  ``nixio.Compression.Auto``: compression as defined during
+   file-opening.
+-  ``nixio.Compression.DeflateNormal``: use compression (fixed level).
+-  ``nixio.CompressionNo``: no compression.
+
+.. code-block:: python
+
+   data_array = b.create_data_array("some data", "nix.sampled", data, compression=nix.Compression.DeflateNormal);
```
Comment (mpsonntag) [inline docs/source/storing_data.rst:249]:
```suggestion
   data_array = b.create_data_array("some data", "nix.sampled", data, compression=nixio.Compression.DeflateNormal);
```

File: docs/source/storing_data.rst [hunk 6]
```diff
@@ -0,0 +1,382 @@
...
+
+Note the following:
+
+1. Compression comes with a little cost of read-write performance.
+2. Data compression is fixed once the *DataArray* has been created, it
+   cannot be changed afterwards.
+3. Opening and extending an compressed *DataArray* is easily possible
```
Comment (mpsonntag) [inline docs/source/storing_data.rst:256]:
```suggestion
3. Opening and extending a compressed *DataArray* is easily possible
```

File: docs/source/storing_data.rst [hunk 7]
```diff
@@ -0,0 +1,382 @@
...
+
+
+The DataFrame
+-------------
+
+The DataFrame stores tabular data in a rows-and-columns format. Each Column has a name, a data type, and, optionally, a unit and a definition.
+
+.. literalinclude:: examples/tabulardata.py
+   :lines: 8 - 25
+   :caption: A DataFrame is created by providing the column definition which consists of a name and the data type that should be stored in each column. Units can be added lateron. It is best to use OrderdDicts to make sure, that the column order is as defined (standard dictionaries do not guarantee the order of the items). :download:`example code <examples/tabulardata.py>`
```
Comment (mpsonntag) [inline docs/source/storing_data.rst:331]:
```suggestion
   :caption: A DataFrame is created by providing the column definition which consists of a name and the data type that should be stored in each column. Units can be added later on. It is best to use OrderdDicts to make sure, that the column order is as defined (standard dictionaries do not guarantee the order of the items). :download:`example code <examples/tabulardata.py>`
```

File: docs/source/tagging.rst [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+multiple points or regions, respectively.
+
+The basic idea is that the *Tag* defines the point (and extent) with
+which it refers to points (or regions) in the data. A tag can point to
+several *DataArrays* at once. These are mere links that are stored in
+the list of *references*. The following figure illustrates, how a
+*MultiTag* links two *DataArrays* to create a new construct.
+
+.. figure:: ./images/mtag_concept.png
+   :alt: An MultiTag can link the data stored in several DataArrays.
```
Comment (mpsonntag) [inline docs/source/tagging.rst:22]:
```suggestion
   :alt: A MultiTag can link the data stored in several DataArrays.
```

File: docs/source/tagging.rst [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+
+   *MultiTags* tag several points or regions in referenced *DataArrays*. By binding entities together they can provide context.
+
+Tagging in 1D
```
Comment (mpsonntag) [inline docs/source/tagging.rst:26]:
```suggestion
Tagging in 1-D
```

File: docs/source/tagging.rst [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+.. figure:: ./images/tag1.png
+   :alt: a system's response to a stimulus
+
+   A system’s response to a stimulus that was on for a certain amount of time.
+
+We may want to store:
+
+1. The recorded system’s response before, during, and after the
+   stimulus.
+2. The time span, the stimulus was on.
```
Comment (mpsonntag) [inline docs/source/tagging.rst:41]:
```suggestion
2. The time span the stimulus was on.
```

File: docs/source/tagging.rst [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+
+In the *NIX* data model *Tag* entities are used to tag regions in a
+*DataArray*. For this, the *Tag* refers to the *DataArray* and stores
+**position** and **extent** of the highlighted segment.
+
+.. figure:: ./images/tag2.png
+   :alt: a system's response to a stimulus
+
+   The system's response is stored in a *DataArray* we do need to add information about the start position and the extent of the stimulus-on segment. The extent is optional, if not specified. the *Tag*/*MultiTag* tags only a point in the data.
```
Comment (mpsonntag) [inline docs/source/tagging.rst:50]:
```suggestion
   The system's response is stored in a *DataArray*, we do need to add information about the start position and the extent of the stimulus-on segment. The extent is optional, if not specified. the *Tag*/*MultiTag* tags only a point in the data.
```

File: docs/source/tagging.rst [hunk 12]
```diff
@@ -0,0 +1,173 @@
...
+
+The following code snippet shows how to create the *Tag*.
+
+.. literalinclude:: examples/tagging_example.py
+    :caption: Creating a Tag for a time span in recorded data :download:`example code <examples/tagging_example.py>`
+    :lines: 78 - 90 
+
+Line 11 creates the *Tag* with the start position and in line 12 we add the extent of the tagged segment. In line 13 we add the *DataArray* we want to point at to the list of *references*. A Tag can refer to more than just a single *DataArray*. Both position and extent are passed as lists with one entry for each dimension of the data. If the length of e.g. position is less than number of dimensions in the referenced *DataArray* it is assumed that the tag refers to the full extent of the non-restricted dimensions. The same applies to missing extents. For the sake of clearness it is advisable to fully specify the positions and extents.
```
Comment (mpsonntag) [inline docs/source/tagging.rst:58]:
```suggestion
Line 11 creates the *Tag* with the start position and in line 12 we add the extent of the tagged segment. In line 13 we add the *DataArray* we want to point at to the list of *references*. A Tag can refer to more than just a single *DataArray*. Both position and extent are passed as lists with one entry for each dimension of the data. If the length of e.g. position is less than the number of dimensions in the referenced *DataArray*, it is assumed that the tag refers to the full extent of the non-restricted dimensions. The same applies to missing extents. For the sake of clearness it is advisable to fully specify the positions and extents.
```

File: docs/source/tagging.rst [hunk 13]
```diff
@@ -0,0 +1,173 @@
...
+ 
+**Note!** Position and extent of the tagged region are given as
+lists. Extent is optional, not setting it tags a point instead
+of a segment of the referenced data.
+
+Tagging in 2D
```
Comment (mpsonntag) [inline docs/source/tagging.rst:64]:
```suggestion
Tagging in 2-D
```

File: docs/source/tagging.rst [hunk 14]
```diff
@@ -0,0 +1,173 @@
...
+-------------
+
+The same principle shown above for 1-D data extends to two or more dimensions. The only difference is that one has to provide vectors of positions/extents whose length matches the number of dimensions of the referenced *DataArray*.
+
+.. figure:: ./images/2d_tag.png
+   :alt: tagging in 2-D
+
+   Tagging works also in 2-D. In this case, position and extent must 2-element vectors (one entry for each dimension).
```
Comment (mpsonntag) [inline docs/source/tagging.rst:72]:
```suggestion
   Tagging works also in 2-D. In this case, position and extent must be 2-element vectors (one entry for each dimension). 
```

File: docs/source/tagging.rst [hunk 15]
```diff
@@ -0,0 +1,173 @@
...
+
+The same principle shown above for 1-D data extends to two or more dimensions. The only difference is that one has to provide vectors of positions/extents whose length matches the number of dimensions of the referenced *DataArray*.
+
+.. figure:: ./images/2d_tag.png
+   :alt: tagging in 2-D
+
+   Tagging works also in 2-D. In this case, position and extent must 2-element vectors (one entry for each dimension). 
+
+
+Tagging multiple points in 1D
```
Comment (mpsonntag) [inline docs/source/tagging.rst:75]:
```suggestion
Tagging multiple points in 1-D
```

File: docs/source/tagging.rst [hunk 16]
```diff
@@ -0,0 +1,173 @@
...
+
+For storing this kind of data we need two *DataArrays*, the first stores the recorded signal, the other the events. Finally, a *MultiTag* entity is used to link both. One can use the event times stored in one of the *DataArrays* to tag multiple points in the other signal.
+
+.. literalinclude:: examples/multiple_points.py
+    :lines: 49-56
+    :caption: To mark several points in a recorded signal we use a *MultiTag* to bind the signal and event times ( :download:`example code <examples/multiple_points.py>`) 
+
+Creating the *MultiTag* is very similar to the creation of the simpler *Tag* above. The main difference is that the tagged positions are not stored in the *MultiTag* itself but we use the event *DataArray* (events in the code example) for this purpose. Finally, the signal *DataArray* is added to the list of references.
+
+Tagging multiple intervals in 1D
```
Comment (mpsonntag) [inline docs/source/tagging.rst:92]:
```suggestion
Tagging multiple intervals in 1-D
```

File: docs/source/tagging.rst [hunk 17]
```diff
@@ -0,0 +1,173 @@
...
+.. figure:: ./images/multiple_regions.png
+   :alt: multiple regions
+
+   With a *MultiTag* we can also tag multiple regions in a signal.
+
+For storing such data we again need one *DataArray* to store the recorded signal. Storing the regions is similar to the approach for the simpler *Tag*, i.e. *positions* and the *extents* need to be provided. Accordingly, **two** additional *DataArray*\ s are required. The first of which stores the positions and the second the extents of the tagged regions. 
+
+.. literalinclude:: examples/multiple_regions.py
+   :caption: The following code tags multiple regions in a 1-D signal (:download:`example code <examples/multiple_regions.py>`).
+   :lines: 27 - 38, 44-46
```
Comment (mpsonntag) [inline docs/source/tagging.rst:107]:
```suggestion
   :lines: 27 - 38, 45-47 
```

File: docs/source/tagging.rst [hunk 18]
```diff
@@ -0,0 +1,173 @@
...
+   :lines: 27 - 38, 44-46 
+
+The example code is rather straight forward. The *DataArrays* ‘positions’ and ‘extents’ take respective data and are added to the created *MultiTag* entity. Finally, the *DataArray* in which we tag the regions is added to the list of *references* of the *MultiTag*.
+
+Tagging in n-D
+--------------
+
+The same principle as demonstrated above applies also to n-dimensional data. Tagging in n dimensions requires **positions** and **extents** stored in *DataArrays* of appropriate shapes.
+
+The following figures show the tagging of multiple regions in 2- and 3D.
```
Comment (mpsonntag) [inline docs/source/tagging.rst:116]:
```suggestion
The following figures show the tagging of multiple regions in 2- and 3-D.
```

File: docs/source/tagging.rst [hunk 19]
```diff
@@ -0,0 +1,173 @@
...
+
+.. figure:: ./images/2d_mtag.png
+   :alt: multiple regions in 2D
```
Comment (mpsonntag) [inline docs/source/tagging.rst:119]:
```suggestion
   :alt: multiple regions in 2-D
```

File: docs/source/tagging.rst [hunk 20]
```diff
@@ -0,0 +1,173 @@
...
+
+   Tagging multiple regions in n-D data requires the *DataArrays* for storing positions and extents to be two-dimensional. The first dimension represents the number of regions, the second has as many entries as the referenced data.
+ 
+According to the number of dimensions of the data (here, width and height) each  starting point and the extent of a tagged region is defined by two numbers. Thus, the **position** and **extent** *DataArrays* are two dimensional. The first dimension represents the number of tagged regions, the second the number of dimensions.
+
+This approach can be extended into n-D. The following figure illustrates the 3-D case.
+
+.. figure:: ./images/3d_mtag.png
+   :alt: multiple regions in 3D
```
Comment (mpsonntag) [inline docs/source/tagging.rst:128]:
```suggestion
   :alt: multiple regions in 3-D
```

File: docs/source/tagging.rst [hunk 21]
```diff
@@ -0,0 +1,173 @@
...
+
+   multiple_regions_3D_plot
+
+The only things that need to be changed in the above code, are the layout of the data (now 3-dimensional) and further entries into **position** and **extent** *DataArrays* along the second dimension (compare tables in the figure). Again, these *DataArrays* are **always** 2D, the first dimension represents the number of tagged regions, the second the number of dimensions.
```
Comment (mpsonntag) [inline docs/source/tagging.rst:132]:
```suggestion
The only things that need to be changed in the above code, are the layout of the data (now 3-dimensional) and further entries into **position** and **extent** *DataArrays* along the second dimension (compare tables in the figure). Again, these *DataArrays* are **always** 2-D, the first dimension represents the number of tagged regions, the second the number of dimensions.
```

**Review by mpsonntag — DISMISSED** [review]
Amazing pull request! Added minor comments only.

**Comment by jgrewe** [comment]
thanks @mpsonntag for the review with super keen eyes!

**Pull request merged**